lib/faraday/adapter/net_http.rb

```diff
@@ -75,9 +75,43 @@ def create_request(env)
       end
 
       def perform_request(http, env)
+        if env[:request].stream_response?
+          size = 0
+          yielded = false
+          http_response = perform_request_with_wrapped_block(http, env) do |chunk|
+            if chunk.size > 0 || size > 0
+              yielded = true
+              size += chunk.bytesize
+              env[:request].on_data.call(chunk, size)
```

Comment:
I think it should be total byte size too because people can get chunk bytesize by `chunk.bytesize`.